MAVProxy/modules/mavproxy_paramedit/param_editor.py

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python
+'''
+param editor module
+Akshath Singhal
+June 2019
+'''
+
+from MAVProxy.modules.lib import mp_util
+from MAVProxy.modules.lib import multiproc
+from MAVProxy.modules.mavproxy_paramedit import ph_event
+import time
+import threading
+from MAVProxy.modules.lib import wx_processguard
+from ..lib.wx_loader import wx
+from MAVProxy.modules.mavproxy_paramedit import param_editor_frame
+from pymavlink import mavutil
+
+ParamEditorEvent = ph_event.ParamEditorEvent
+
+
+class ParamEditorEventThread(threading.Thread):
+    def __init__(self, mp_paramedit, queue, lock):
+        threading.Thread.__init__(self)
+        self.mp_paramedit = mp_paramedit
+        self.event_queue = queue
+        self.event_queue_lock = lock
+        self.time_to_quit = False
+        self.queue_access_timeout = 0.5
+
+    def module(self, name):
+        '''access another module'''
+        return self.mp_paramedit.mpstate.module(name)
+
+    def master(self):
+        '''access master mavlink connection'''
+        return self.mp_paramedit.mpstate.master()
+
+    def run(self):
+        while not self.time_to_quit:
+            queue_access_start_time = time.time()
+
+            while (self.event_queue.qsize() > 0):
+                event = self.event_queue.get(block=False)
+                event_type = event.get_type()
+
+                if event_type == ph_event.PEE_READ_PARAM:
+                    self.param_received = self.module('param').mav_param
+                    self.mp_paramedit.gui_event_queue_lock.acquire()
```

Comment:
We really shouldn't need to be doing our own locking here - it's a multiprocessing queue which should be thread-safe!.  But it seems to me our MacOSX workarounds may not behave as desired.

Reply:
I have made the changes but I dont have a MacOSX to check if it work fine.